hail/python/hail/vds/combiner/variant_dataset_combiner.py

```diff
@@ -50,34 +65,146 @@ def read_variant_datasets(inputs: List[str], intervals: List[Interval], interval
+    save_path : :class:`str`
+        The location to store this VariantDatasetCombiner plan. A failed execution can be restarted
+        using this plan.
+    output_path : :class:`str`
+        The location to store the new VariantDataset.
+    temp_path : :class:`str`
+        The location to store temporary intermediates. We recommend using a bucket with an automatic
+        deletion or lifecycle policy.
+    reference_genome : :class:`.ReferenceGenome`
+        The reference genome to which all inputs (gVCFs and Variant Datasets) are aligned.
+    branch_factor : :class:`int`
+        The number of Variant Datasets to combine at once.
+    target_records : :class:`int`
+        The target number of variants per partition.
+    gvcf_batch_size : :class:`int`
+        The number of gVCFs to combine into a Variant Dataset at once.
+    contig_recoding : :class:`dict` mapping :class:`str` to :class:`str` or :obj:`None`
+        This mapping is applied to gVCF contigs before importing them into Hail. This is used to
+        handle gVCFs containing invalid contig names. For example, GRCh38 gVCFs which contain the
+        contig "1" instead of the correct "chr1".
+    vdses : :class:`list` of :class:`.VDSMetadata`
+        A list of Variant Datasets to combine. Each dataset is identified by a
+        :class:`.VDSMetadata`, which is a pair of a path and the number of samples in said Variant
+        Dataset.
+    gvcfs : :class:`list` of :class:`.str`
+        A list of paths of gVCF files to combine.
+    gvcf_sample_names : :class:`list` of :class:`str` or :obj:`None`
+        List of names to use for the samples from the gVCF files. Must be the same length as
+        `gvcfs`. Must be specified if `gvcf_external_header` is specified.
+    gvcf_external_header : :class:`str` or :obj:`None`
+        A path to a file containing a VCF header which is applied to all gVCFs. Must be specified if
+        `gvcf_sample_names` is specified.
+    gvcf_import_intervals : :class:`list` of :class:`.Interval`
+        A list of intervals defining how to partition the gVCF files. The same partitioning is used
+        for all gVCF files. Finer partitioning yields more parallelism but less work per task.
+    gvcf_info_to_keep : :class:`list` of :class:`str` or :obj:`None`
+        gVCF ``INFO`` fields to keep in the ``gvcf_info`` entry field. By default, all fields are
+        kept except ``END`` and ``DP`` are kept.
+    gvcf_reference_entry_fields_to_keep : :class:`list` of :class:`str` or :obj:`None`
+        Genotype fields to keep in the reference table. If empty, the first 10,000 reference block
+        rows of ``mt`` will be sampled and all fields found to be defined other than ``GT``, ``AD``,
+        and ``PL`` will be entry fields in the resulting reference matrix in the dataset.
+
+    """
+    _default_gvcf_batch_size = 100
+    _default_branch_factor = 100
+    _default_target_records = 24_000
+    _gvcf_merge_task_limit = 150_000
```

Comment:
prefixing attributes with an underscore hides them from the docs and avoids "missing docstring" warnings.

Reply:
great.